### DISCUSSION TITLE
What is the future for Open Orienteering Mapper

### DISCUSSION THREAD

**Discussion by ohartvic** [OpenOrienteering/mapper#2149] [General]
I assume someone must have already asked the question - What is the future development pan like? 

I love using Open Orienteering Mapper, but since there was not a release over the last two years I wonder is this the end of a wonderful tool? I sincerely hope not but I would like to hear from you guys what you think and/or plan? 

I would be ready to regularly financialy contribute to keep OOM alive. 

Thank you for your answer.

**Comment by dg0yt** [comment]
I hope I can spent more time directly on Mapper soon. The last two years were also used for activities which should eventually benefit the Mapper software, not just other projects.

**Reply by ohartvic** [reply]
Thank you for your anwser, looking forward to it.

**Comment by jmacura** [comment]
> I would be ready to regularly financialy contribute to keep OOM alive.
> 

This point directs me into a follow-up question. Have you considered any possibility of funding the project via donations? I can imagine some satisfied users might be willing to donate. And nowadays platforms (even GitHub itself) make it pretty easy technically. 
Though I understand that it also brings some administrative burden. So I am interested in your thoughts (you, the maintainers).

**Comment by yevhenmazur** [comment]
In the fall, I organized an o-mapping workshop in Moldova where OpenOrienteering Mapper was used extensively. I recommended that students use the most recent official version, but it turned out to be outdated. It lacks current symbol sets, and magnetic declination acquisition (a part of the seminar demonstration) does not work.

Today a colleague from Italy asked me about problems opening OCAD files. I advised him to try the development release, but I can't find a link to it. For my own use, I build OOM from the source code, but I cannot recommend it to other users.

Due to a significant delay in the release, the project starts to lose its own value and gets a reputation as an unsupported product. I would like to stop this loss of value and am also ready to support the project financially or in another relevant way.

**Reply by oiLvAcciNe** [reply]
Hi,
Can you explain the process of building a .exe/.msi from the source code?
tks

**Reply by jjojala** [reply]
See [INSTALL.md](https://github.com/OpenOrienteering/mapper/blob/master/INSTALL.md). Just went trough [Compiling for windows without superbuild](https://github.com/OpenOrienteering/mapper/blob/master/INSTALL.md#compiling-for-windows-without-openorienteering-superbuild). Mostly up-to-date, but in addition to instructions I had to to install qt5 and something else too (gcc?).

**Comment by D3lta0300** [comment]
@dg0yt  I am trying to convince my federation (FFCO) to fully invest in OOM instead of OCAD. You have done a great work, and it would nice if it becomes the best mapping software available. The fact that you let your work open source is amazing !
How much time do you think is needed in order to make OOM greater than anything else ? How much would it cost ?

**Reply by dg0yt** [reply]
> How much time do you think is needed in order to make OOM greater than anything else ? How much would it cost ?

I guess it could cost twice as much as OCAD.

IMO both the commercial and the community offer should be equally appreciated and respected. They have individual strengths and weaknesses.

**Comment by kjetilk** [comment]
Is there potential for a maintenance release soon? The main reason why I ask is that Ubuntu freezes on the 29th of February for their next LTS (Long Term Support) release. I realize that when a project is in a lot of flux, frozen packages may not be so desirable from a developer perspective, but now, the Mapper appears to me as both mature and stable.

And so, if there is anything that is in the repo but not yet released that has low risk, might even make things cleaner, then it sounds to me as a good idea to release it, perhaps even as a 1.0, so that the Debian and Ubuntu packages can be used some time into the future. It doesn't have to be big.

The Debian maintainers would need a bit of time to update the packages, so it would need to be done some time before that to be useful for this, but I'd like to encourage a small review for this.

**Comment by pkturner** [comment]
I agree that that the lack of a release is a substantial impediment to Mapper, both to users and to developers.

There has been enough progress on the master branch since v0.9.5 to make a fresh pre-release worthwhile. Perhaps the obsolescence of the “latest” superbuild is holding up creation of a pre-release. I wonder how difficult it would be to bring the superbuild up to date.

**Comment by lpechacek** [comment]
While my announcement is tangential to the question of the OpenOrienteering project future, I believe that I can address the question about the Mapper future development and maintenance releases.

I've started a fork called LibreMapper at https://github.com/LibreMapper/mapper. I'm busy with the rebranding now. As soon as I complete the fork, I'll merge the fixes that accumulated in the pull requests here and make a release as soon as I manage to produce Windows and Android builds. Then I'll merge further enhancements and turn my focus on code cleanup and hardening.  Let's see how far I get with this initiative.

If someone wants to help me with a rework of the CI/CD, I'll be grateful for the assistance.

![Screenshot from 2024-04-02 23-31-04](https://github.com/OpenOrienteering/mapper/assets/5584406/cf889da6-5194-4500-a9c7-b2b010de3dd8)

**Reply by jmacura** [reply]
I feel a strong dichotomy in my brain. On one hand I am sad that the team behind OO Mapper is breaking into fractions and individual playgrounds. On the other hand, I am grateful for active development and a promise of new releases.
Anyway... Thank you @lpechacek !

**Reply by lpechacek** [reply]
> the team behind OO Mapper is breaking into fractions and individual playgrounds

I beg to disagree that the team is breaking. I'm not going anywhere and there are three more guys working on the software. At this stage, we can look at LibreMapper as an OO Mapper pre-release or an experimental release that is not managed by OpenOrienteering.

**Reply by jjojala** [reply]
What is the reason, why this (== new release) is not made within the original OpenOrienteering/Mapper project - especially if the goal is just to launch a new release? If the developers disagree the overall goals or directions of the project, splitting is understandable (though not necessarily desirable). This doesn't seem to be the case, or is it? Disagreement on when to release alone sounds to me a bad reason to fork the project?

**Reply by lpechacek** [reply]
> What is the reason, why this (== new release) is not made within the original OpenOrienteering/Mapper project - especially if the goal is just to launch a new release?

Please mind the "at this stage" qualification. That is, the first LibreMapper release will be very close to OO Mapper. Later LibreMapper releases will diverge. But we are not yet that far. That said, the goal is not a Mapper release.

Regarding project philosophy, I tried to summarize my position in [LibreMapper's Contributing.md](https://github.com/LibreMapper/mapper/blob/main/CONTRIBUTING.md) document. In short, OO Mapper strives for stability and maintainability. LibreMapper won't.

**Reply by ollesmaps** [reply]
Thank you Libor for your continuous support of the mapping community. Lets see where LibreMapper will take us.
On a side note. Would you take the unresolved issues from OOMapper to LibreMapper?

**Reply by lpechacek** [reply]
> On a side note. Would you take the unresolved issues from OOMapper to LibreMapper?

Jein. There are different types of issue tickets, much like there are different types of pull requests. So, defect reports applicable to LibreMapper will see a corresponding fix in the code. Enhancement requests towards OO Mapper may not be reflected in LibreMapper, however. Mind that LibreMapper is not meant to be a clone of neither OO Mapper nor OCAD. There should be a genuine fun element in the project. That fun part might, be killed by hunting after someone else's goals which I'd rather avoid. As for pull requests, about 25 % of the pending ones have been merged into LibreMapper as of now. Mainly OCD compatibility enhancements and bug fixes.

A terse list of future steps is at https://github.com/orgs/LibreMapper/projects . It's too brief as I'm not that familiar with the advanced GH Projects tool and cannot fill in a more detailed description into the task tickets. But reading through it should give an idea about where we are heading mid-term.

**Comment by jjojala** [comment]
Hi, I am a bit worried about the future about the Mapper due the lack of recent releases! I really feel that this hinders new users to apply Mapper and it eventually starts deteriorate current user base as well! From a non-sw-developer's point of view this project may sound dead! A whole lot of bug fixes and improvements have been implemented since 0.9.5 (four years ago). Why not release?

**Comment by dl3sdo** [comment]
As Mapper is coming closer to 0.9.6 please help improving the [translations](https://github.com/OpenOrienteering/mapper#translating). Since Tuesday there are round about 50 words and phrases which are not yet translated (except Czech, French and German).

**Reply by jmacura** [reply]
The latest stable builds are kind of hard to find... Since they are no longer present in the GH releases (understandable), also the links for "master" and "unstable" builds from https://www.openorienteering.org/apps/mapper/ are pointing just to the 0.9.5 release.
I am willing to try the latest "nightly/dev/unstable" builds more often, but I always struggle finding them. Isn't there a way to make them easier to find for the testers / bold users?

**Reply by dg0yt** [reply]
No. The plan is to make a release.

**Reply by lpechacek** [reply]
@dg0yt , I sense two questions in Jan's inquiry. One of them is about availability of development builds. The other, implicit, is about involvement of volunteer testers in the development. Does the 'no' apply to both parts or just the availability of pre-release packages?

**Reply by dg0yt** [reply]
I can't speculate on the questions behind the questions. But "easy to find" already maps correctly to what is available for the public.
This does include master and unstable builds for Linux from build.opensuse.org BTW.

**Reply by lpechacek** [reply]
Thanks, Kai! That clarifies the intentions for me.

**Reply by jjojala** [reply]
What I actually meant, was that how the Mapper is shown for the ordinary users. Most of them are likely unwilling and/or incapable to find test- and nightly builds (if they do know that such things exists/are available!). They look at https://www.openorienteering.org/ and found 0.9.5 from March 2021. For them it looks like no progress.

Maybe the lack of "stable releases" is not so big problem after all, if test and bleeding edge nightly builds would have shortly explained and made available through www.openorienteering.org? That would at least make obvious, that the progress is not stopped and would give "ordinary users" a change to use newer packages.

**Reply by dl3sdo** [reply]
Maybe it helps to create a description issue where to find and how to download the latest stable build, that issue could be sticked to the 'Issues' page.

**Reply by dg0yt** [reply]
@dl3sdo No. There are obligations attached to distributing this software.
Throwing out more unofficial builds (tagged stable or not) is not the way to go.

**Reply by ohartvic** [reply]
happy to wait as long as it is ensured the OMapper is not going to disappear :)

**Reply by ollesmaps** [reply]
Thanks all the volunteers for their work they put into Open Orienteering Mapper. It is a great tool and pushes the sport of orienteering forward. Thanks for the newest realease 0.9.6!
